pkg/controller/podautoscaler/controller_metrics/metrics.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// metrics packages contains metrics which are exposed from the HPA controller.
+package metrics
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+type ReconciliationAction string
```

Comment:
Since all recordings come from `MetricsRecorder` I would:

* move all public names like `ReconciliationAction` to recorder file.
* lowercase all private names like `HPAControllerSubsystem` and prometheus metric vars